nightfall-deployer/contracts/Shield.sol

```diff
@@ -49,6 +49,14 @@ contract Shield is Stateful, Config, ReentrancyGuardUpgradeable, Pausable, KYC {
                 payIn(t);
             }
         }
+
+        uint256 maxBlockSize = MAX_BLOCK_SIZE;
```

Comment:
I think this must be moved up - possibly above Line 44, otherwise I think the `payIn` will still occur and 1) we will take people's funds even though it will error and 2) it will trigger an escrow of the funds which means a deposit can then be sent off-chain and bypass this check.

Reply:
But, isn't the transaction atomic? Meaning if it fails everything is reverted? Anyway, I think it is better to have it prior the loop